Core/include/Acts/EventData/SpacePointContainer.hpp

```diff
@@ -0,0 +1,239 @@
+							 true>;
+    
+  public:    
+    // Constructors
+    // It makes sense to support both options of
+    // taking or not the ownership
+
+    // Do not take ownership
+    // Activate only if holder_t is RefHolder
+    template<template <typename> class H = holder_t,
+             std::enable_if_t<
+               Acts::detail_tc::is_same_template<H, Acts::detail_tc::RefHolder>::value
+               >* = nullptr>
+    SpacePointContainer(container_t& container)
+      : m_container( container )
+    {
+      IndexType n = size();
+      m_proxies.reserve(n);
+      for (IndexType i(0); i<n; i++)
+        m_proxies.emplace_back( *this, i );
+    }
+    
+    // Take the ownership
+    // Activate only if holder_t is ValueHolder
+    template<template <typename> class H = holder_t,
+             std::enable_if_t<
+               Acts::detail_tc::is_same_template<H, Acts::detail_tc::ValueHolder>::value
+               >* = nullptr>
+    SpacePointContainer(container_t&& container)
+      :	m_container( container )
+    {
+      IndexType n = size();
+      m_proxies.reserve(n);
+      for (IndexType i(0); i<n; i++)
+        m_proxies.emplace_back( *this, i );
+    }
+    
+    // If we take ownership, forbid copy operations
+    // Need to define copy operations only if holder_t is RefHolder !!!
+    template<template <typename> class H = holder_t,
+	     std::enable_if_t<
+	       Acts::detail_tc::is_same_template<H, Acts::detail_tc::RefHolder>::value
+	       >* = nullptr>
+    SpacePointContainer(SpacePointContainer& other)
+      : m_container( *m_container.ptr ),
+	m_proxies( other.m_proxies.begin(), other.m_proxies.end() )
+    {}
+
+    template<template <typename> class H = holder_t,
+	     std::enable_if_t<Acts::detail_tc::is_same_template<H, Acts::detail_tc::RefHolder>::value, bool> = true>
+    SpacePointContainer& operator=(SpacePointContainer& other)
+    {
+      m_container.ptr = other.m_container.ptr;
+      m_proxies.insert( m_proxies.end(), other.m_proxies.begin(), other.m_proxies.end() );
+      return *this;
+    }
+    
+    // move operations
+    SpacePointContainer(SpacePointContainer&& other) noexcept
+      : m_container( std::exchange( other.m_container.ptr, nullptr) ),
+	m_proxies( std::move(other.m_proxies) )
+    {}
+
+    SpacePointContainer& operator=(SpacePointContainer&& other) noexcept
+    {
+      m_container = std::exchange( other.m_container.ptr, nullptr);
+      m_proxies = std::move( other.m_proxies );
+      return *this;
+    }
+    
+    // Destructor
+    ~SpacePointContainer() = default;
+    
+    IndexType size() const;
+
+    iterator begin();
+    iterator end();
+    
+    const_iterator begin() const;
+    const_iterator end() const;
+
+    SpacePointProxyType& get(IndexType n);
+    const SpacePointProxyType& get(IndexType n) const;
+
+    // do these need to be private or public?
+    const container_t& container() const;
+
```

Comment:
Would we also need method allowing to construct this container, insert?

Do we ever need to sort the SPs container? What is needed for it?